include/ginkgo/core/base/exception.hpp

```diff
@@ -269,6 +269,33 @@ class DimensionMismatch : public Error {
 };
 
 
+/**
+ * BadDimension is thrown if an operation is being applied to a LinOp of
+ * with bad dimensions.
+ */
+class BadDimension : public Error {
+public:
+    /**
+     * Initializes a bad dimension error.
+     * @param file The name of the offending source file
+     * @param line The source code line number where the error occurred
+     * @param func The function name where the error occurred
+     * @param first_name The name of the first operator
+     * @param first_rows The output dimension of the first operator
+     * @param first_cols The input dimension of the first operator
```

Comment:
Why state that it is the `first` when it is the only parameter with that name? All three names are not really descriptive. I would suggest:
 first_name -> operator_name / variable_name
first_row -> number_rows
first_cols -> number_cols

With appropriate descriptions.